### DISCUSSION TITLE
Is there documentation on clap_event_transport

### DISCUSSION THREAD

**Discussion by sagantech** [free-audio/clap#164] [Q&A] [answered]
Specifically there is this in events.h:

clap_beattime song_pos_beats;   // position in beats

clap_beattime  is defined as int64_t. So song_pos_beats is a 64 bit integer. What are the units of measurement? With VST2 it was a double and at least documented as "Musical Position, in Quarter Note (1.0 equals 1 Quarter Note)"

Can anyone help me here?

**Comment by robbert-vdh** [comment, marked as answer]
There's a comment on how to use these fixed point numbers just above that definition for `clap_beattime`:

https://github.com/free-audio/clap/blob/af1e05ce6487151c6b36003fdb47566dbb50194e/include/clap/fixedpoint.h#L6-L16

And the length of a beat is determined by the tempo. So at a tempo of 174 beats per minute, a beat takes `60 / 174 ≈ 0.345` seconds. VST2 and VST3 use the term quarter note here, but I'm fairly sure they all refer to the same thing.

**Reply by sagantech** [reply]
Thanks I think I understand now. I didn't realize that this was a fixed point integer.

**Reply by baconpaul** [reply]
If it helps, <https://github.com/free-audio/clap-juce-extensions/blob/698e0d05f5a2ac798c8bb30cfceeba5c4cdf45bf/src/wrapper/clap-juce-wrapper.cpp#L454> shows how we bridge the clap data structure to the juce TransportInfo object.

**Reply by sagantech** [reply]
Yes that is helpful. Thanks.
